Lib/http/server.py

```diff
@@ -635,18 +695,31 @@ class SimpleHTTPRequestHandler(BaseHTTPRequestHandler):
 
     server_version = "SimpleHTTP/" + __version__
 
+    # List of Content Types that are returned with HTTP compression (gzip).
+    # Set to the empty list by default (no compression).
+    compressed_types = []
+
     def __init__(self, *args, directory=None, **kwargs):
         if directory is None:
             directory = os.getcwd()
         self.directory = directory
+        self.chunked = False
         super().__init__(*args, **kwargs)
 
     def do_GET(self):
         """Serve a GET request."""
         f = self.send_head()
         if f:
             try:
-                self.copyfile(f, self.wfile)
+                if not self.chunked:
+                    self.copyfile(f, self.wfile)
+                else:
+                    # Chunked Transfer Encoding (RFC 7230 section 4.1)
+                    # f is a generator of chunks. The last chunk is b''.
+                    for chunk in f:
+                        length = hex(len(chunk)).encode("ascii")
```

Comment:
I agree it would be better to share code with the chunk encoder in _http.client_, but I couldn't find a simple way to do it, and producing chunks is a very short piece of code.

I replaced `hex()` by `"{:X}".format()` ; I didn't see the bug because _http.client_ ignores the leading `0x`